packages/mui-base/src/TextareaAutosize/TextareaAutosize.tsx

```diff
@@ -94,8 +133,8 @@ const TextareaAutosize = React.forwardRef(function TextareaAutosize(props, ref)
     return { outerHeightStyle, overflow };
   }, [maxRows, minRows, props.placeholder]);
 
-  const updateState = (prevState, newState) => {
-    const { outerHeightStyle, overflow } = newState;
+  const updateState = (prevState: State, newState: State) => {
+    const { outerHeightStyle = 0, overflow } = newState;
```

Comment:
Maybe to move to the state initialization, this should never be undefined otherwise.

```suggestion
    const { outerHeightStyle, overflow } = newState;
```